pkg/ocp/ocp.go

```diff
@@ -19,62 +23,66 @@ type ConfigFile struct {
 	Content  []byte
 }
 
-type ConfigMaster struct {
+type OAuthConfig struct {
 	ConfigFile
-	OCP3 ocp3.Master
-	OCP4 ocp4.Master
+	OCP3    configv1.MasterConfig
+	OAuth   oauth.OAuthCRD
+	Secrets []secrets.Secret
 }
 
-type ConfigNode struct {
+type SDNConfig struct {
 	ConfigFile
-	OCP3 ocp3.Node
-	OCP4 ocp4.Node
+	OCP3 configv1.MasterNetworkConfig
+	SDN  sdn.NetworkCR
 }
 
 type Translator interface {
 	Add(string)
-	Decode()
-	Fetch(string)
+	Extract()
+	Transform()
 	GenYAML() ocp4.Manifests
```

Comment:
This shouldn't be part of this interface. Semantically it doesn't make any sense on a `Translator` type, and the fact that it's `YAML` is an implementation detail.

Furthermore, it's a common practice with go to keep interfaces small and compose them. For example, see go's own [ReadWriter](https://golang.org/pkg/io/#ReadWriter).

There is a dividing line. It's not useful to take each of these methods and put them on individual interfaces. The atomic type here is the `Translator`. It's not useful to break it down any further.